src/www/ui/api/Controllers/UserController.php

```diff
@@ -176,13 +183,14 @@ public function getCurrentUser($request, $response, $args)
    */
   public function updateUser($request, $response, $args)
   {
-    $id = intval($args['id']);
+    $apiVersion = ApiVersion::getVersion($request);
+    $id = $apiVersion == ApiVersion::V2 ? intval($this->restHelper->getUserDao()->getUserByName($args['pathParam'])['user_pk']) : intval($args['pathParam']);
     if ($this->dbHelper->doesIdExist("users","user_pk", $id)) {
```

Comment:
Oops, think we never tested it. This condition needs to be inverted (user not exists) to throw the exception. Can you please do that?

```suggestion
    if (!$this->dbHelper->doesIdExist("users","user_pk", $id)) {
```

Reply:
Ah, I believe I overlooked that detail during testing as well. Fixed